test/unittest/xmlparser/XMLParserTests.cpp

```diff
@@ -617,6 +661,2013 @@ TEST_F(XMLParserTests, DataBuffer)
+        ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+        titleElement = xml_doc.RootElement();
+        EXPECT_EQ(XMLP_ret::XML_OK, XMLParserTest::parse_tls_config_wrapper(titleElement, tcp_transport));
+        std::shared_ptr<TCPTransportDescriptor> descriptor =
+                std::dynamic_pointer_cast<TCPTransportDescriptor>(tcp_transport);
+        EXPECT_TRUE(
+            descriptor->tls_config.get_option(TCPTransportDescriptor::TLSConfig::TLSOptions::SINGLE_DH_USE));
+    }
+}
+
+/*
+ * This test checks the correct functioning of the parseProfile function for all xml child elements of <profile>.
+ * 1. Check that elements <transport_descriptors>, <library_settings>, <participant>, <publisher>, <data_writer>,
+ * <subscriber>, <data_reader>, <topic>, <requester>, <replier>, and <types> are read as xml child elements of the
+ * <profiles> root element.
+ * 2. Check that it triggers an error when reading an wrong element.
+ */
+TEST_F(XMLParserTests, parseProfiles)
+{
+    tinyxml2::XMLDocument xml_doc;
+    tinyxml2::XMLElement* titleElement;
+    std::unique_ptr<BaseNode> profilesNode;
+
+    // Parametrized XML
+    const char* xml_p =
+            "\
+            <profiles>\
+                <%s>\
+                </%s>\
+            </profiles>\
+            ";
+
+    char xml[600];
+
+    std::vector<std::string> elements_ok {
+        "transport_descriptors",
+        "publisher",
+        "data_writer",
+        "subscriber",
+        "data_reader",
+        "topic",
+        "types"
+        // "qos_profile",
+        // "application",
+        // "type"
+    };
+    for (std::string e : elements_ok)
+    {
+        sprintf(xml, xml_p, e.c_str(), e.c_str());
+        ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+        titleElement = xml_doc.RootElement();
+        profilesNode.reset(new BaseNode{ NodeType::PROFILES });
+        EXPECT_EQ(XMLP_ret::XML_OK, XMLParserTest::parseProfiles_wrapper(titleElement, *profilesNode));
+    }
+
+    std::vector<std::string> elements_error {
+        "library_settings",
+        "participant",
+        "requester",
+        "replier",
+        "bad_element"
+    };
+    for (std::string e : elements_error)
+    {
+        sprintf(xml, xml_p, e.c_str(), e.c_str());
+        ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+        titleElement = xml_doc.RootElement();
+        profilesNode.reset(new BaseNode{ NodeType::PROFILES });
+        EXPECT_EQ(XMLP_ret::XML_ERROR, XMLParserTest::parseProfiles_wrapper(titleElement, *profilesNode));
+    }
+}
+
+/*
+ * This test checks the correct functioning of the parseProfile function for all unsupported xml child elements of
+ * <profile>.
+ * 1. Check that unsupported elements <qos_profile>, <application>, and <type> are read as xml child elements of the
+ * <profiles> root element.
+ * 2. Check that it outputs a Log Error when reading an unsupported element.
```

Comment:
Some tests do this check and some don't

Reply:
Only test which produce log errors but do not return an error code. Only for unsupported tags where the rest of the XML is correct.